### DISCUSSION TITLE
XPRA over TCP vs SSH

### DISCUSSION THREAD

**Discussion by louwe** [Xpra-org/xpra#3852] [Q&A] [answered]
Hi, I use XPRA usually either over a secure LAN or tunneled over Wireguard. As such, I'm not worried about encrypting the connection and just want to have the best and fastest performance possible. In that case, is it better to just use plain TCP vs the usual recommended SSH method?

Thanks!

**Comment by totaam** [comment, marked as answer]
For sure `tcp` mode.
There are some unusual corner cases where `ssh` mode can save a little bit of bandwidth because of the extra buffering that it does, but this costs some latency and it should be very rare.

**Reply by louwe** [reply]
Thank you for the prompt and clear answer!

**Comment by totaam** [comment]
In xpra v5, the quic transport (#3376) could be used to for speed things up but it will probably take time before xpra can fully take advantage of it.

**Reply by louwe** [reply]
Nice heads-up. I'll be sure to look forward to it!
